pypsa/statistics/expressions.py

```diff
@@ -1213,7 +1354,7 @@ def func(n: Network, c: str, port: str) -> pd.Series:
                 "spill_cost",
             ]:
                 if cost_type in cost_types_ and cost_type in n.static(c):
-                    attr = lookup.query(cost_type).loc[c].index.item()
+                    attr = lookup.query(cost_type).loc[c].index.item() + port
```

Comment:
Good catch! That's for links with `p0`, right? The important thing is that any `p1` or so would make this calculation invalid. Is that caught?  That's the difference between `statistics/expressions.py` and `optimization/expressions.py` :)